### PR TITLE
WT-9954 Fix bulk cursors and tiered checkpoints. Enable tests.

### PR DESCRIPTION
**Pull request by sueloverso**
@keitharnoldsmith  there is code a bit further down that also skips but for this case of bulk cursors it is too late. That code was added in WT-10989 which is a large and complex ticket you and Sulabh were involved in. I considered removing that code but did not have the context and didn't feel it was safe. If you understand the scenarios better and feel the old code can be removed let me know.

### CODE REVIEW

File: src/txn/txn_ckpt.c [hunk 1]
```diff
@@ -138,18 +140,27 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
                 }
             }
             /* Only instantiate the handle if we need to flush. */
-            WT_ERR(__wt_session_get_dhandle(session, key, NULL, NULL, 0));
+            ret = __wt_session_get_dhandle(session, key, NULL, NULL, 0);
```
Comment (etienneptl) [inline src/txn/txn_ckpt.c:143]: Should we use `WT_ERR_ERROR_OK` instead with `keep` set to `true`?
```suggestion
            WT_ERR_ERROR_OK(__wt_session_get_dhandle(session, key, NULL, NULL, 0), EBUSY, true);
```

File: src/txn/txn_ckpt.c [hunk 2]
```diff
@@ -138,18 +140,27 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
...
+            /* If we get back EBUSY, this handle may be open with bulk or other special flags. We
```
Comment (etienneptl) [inline src/txn/txn_ckpt.c:144]: Comment formatting needs to be fixed.
```suggestion
            /* 
             * If we get back EBUSY, this handle may be open with bulk or other special flags. We
```

File: src/txn/txn_ckpt.c [hunk 3]
```diff
@@ -138,18 +140,27 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
...
+             * need to skip it this tree. We fake checkpoints for them, i.e. we never really write
```
Comment (etienneptl) [inline src/txn/txn_ckpt.c:145]: Is there a typo? Did you mean
```suggestion
             * need to skip this tree. We fake checkpoints for them, i.e. we never really write
```
Or
```suggestion
             * need to skip it for this tree. We fake checkpoints for them, i.e. we never really write
```

File: src/txn/txn_ckpt.c [hunk 4]
```diff
@@ -138,18 +140,27 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
...
+             * a checkpoint to the disk and we cannot get the dhandle now.
+             */
+            if (ret == EBUSY) {
+                WT_STAT_CONN_INCR(session, flush_tier_skipped);
+                continue;
```
Comment (etienneptl) [inline src/txn/txn_ckpt.c:150]: Do we want to reset `ret` to `0` here?

File: src/txn/txn_ckpt.c [hunk 5]
```diff
@@ -138,18 +140,27 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
...
+            } else
```
Comment (etienneptl) [inline src/txn/txn_ckpt.c:151]: We won't need those two lines if we use `WT_ERR_ERROR_OK` as suggested above.

**Review by etienneptl — COMMENTED** [review]
Hi @sueloverso, I was interested in this PR and left a few comments, I hope that's ok.
When you say you considered removing the code from WT-10989, as in reverting?

**Comment by sueloverso** [comment]
@etienneptl thank you for the review. No I did not mean revert 10989. In the same area of code, just a few lines down, in txn_ckpt.c there is a block and comment (that in current develop reads as):
```
            /*
             * When we call wt_tiered_switch the session->dhandle points to the tiered: entry and
             * the arg is the config string that is currently in the metadata. Also, mark the tree
             * dirty to ensure it participates in the checkpoint process, even if clean. Skip the
             * trees still open for bulk insertion, we fake checkpoints for them, i.e. we never
             * really write a checkpoint to the disk - so no point switching just yet.
             */
            btree = S2BT(session);
            if (btree->original) {
                WT_STAT_CONN_INCR(session, flush_tier_skipped);
                WT_ERR(__wt_session_release_dhandle(session));
                continue;
            }
```
It appears that `btree->original` and skipping is referring to the bulk insertion in the comment. But we cannot get to here with a bulk cursor because trying to get the dhandle above returns EBUSY.

File: src/txn/txn_ckpt.c (hunk 1 above)
Comment (sueloverso) [reply src/txn/txn_ckpt.c:143]: Yes! Fixed.

File: src/txn/txn_ckpt.c (hunk 2 above)
Comment (sueloverso) [reply src/txn/txn_ckpt.c:144]: Good catch. Fixed.

**Review by ddanderson — APPROVED** [review]
LGTM

**Review by etienneptl — APPROVED** [review]
LGTM!

**Comment by sueloverso** [comment]
evergreen merge

**Pull request merged**